oceannavigator/frontend/src/components/AreaWindow.jsx

```diff
@@ -98,33 +98,42 @@ export default class AreaWindow extends React.Component {
     this._mounted = false;
   }
 
-  componentWillReceiveProps(props) {
-    if (this._mounted && stringify(this.props) !== stringify(props)) {
-
-      if (props.scale !== this.props.scale) {
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.colormap_diff !== this.state.colormap_diff) {
+      this.forceUpdate()
+    }
+    if (this._mounted && stringify(prevProps) !== stringify(this.props)) {
+      if (prevProps.scale !== this.props.scale) {
         if (this.state.scale.indexOf("auto") !== -1) {
           this.setState({
-            scale: props.scale + ",auto"
+            scale: this.props.scale + ",auto"
           });
         } else {
           this.setState({
-            scale: props.scale,
+            scale: this.props.scale,
           });
         }
       }
 
       // Update time indices
-      if (props.dataset_0.time !== this.state.dataset_0.time) {
+      if (prevProps.dataset_0.time !== this.state.dataset_0.time) {
         this.setState(
           {
-            output_starttime: props.dataset_0.time,
-            output_endtime: props.dataset_0.time
+            output_starttime: this.props.dataset_0.time,
+            output_endtime: this.props.dataset_0.time
           }
         );
       }
     }
   }
 
+  /*componentWillReceiveProps(props) {
+    if (this._mounted && stringify(this.props) !== stringify(props)) {
```

Comment:
If we're not using this code anymore, might as well delete it. If we need it in the future we can look back in the git history

Reply:
Will do

Reply:
> Judging from the change from `componentWillReceiveProps` to `componentDidUpdate`, React wasn't updating the state properly?
> 
> Anyway other than the comment about the unneeded code, it's good to go. Would it also be possible to attach a screenshot of the fixed plot in the original post? I find it helps with constructing a timeline in the event something breaks down the line.

The intention was to fix the reloading issue. It didn't work.
That being said, componentWillReceiveProps throws deprecation warnings, so we should switch over eventually anyway.